pkg/proxy/healthcheck/service_health.go

```diff
@@ -52,9 +52,9 @@ type ServiceHealthServer interface {
 	SyncEndpoints(newEndpoints map[types.NamespacedName]int) error
 }
 
-func newServiceHealthServer(hostname string, recorder events.EventRecorder, listener listener, factory httpServerFactory, nodePortAddresses []string) ServiceHealthServer {
+func newServiceHealthServer(hostname string, recorder events.EventRecorder, listener listener, factory httpServerFactory, nodePortAddresses *utilproxy.NodePortAddresses) ServiceHealthServer {
 
-	nodeAddresses, err := utilproxy.GetNodeAddresses(nodePortAddresses, utilproxy.RealNetwork{})
+	nodeAddresses, err := nodePortAddresses.GetNodeAddresses(utilproxy.RealNetwork{})
```

Comment:
actually the serviceHealthServer looks wrong now, since only uses the addresses that are obtained here, if those addresses change over time they will not be reflected ... maybe `utilproxy.NodePortAddresses` should be part of the `server` type and then return the current addresses?

Reply:
> looks wrong now

The behavior is unchanged from before though, so if it's wrong now it was always wrong. The health server just opens one or more listening sockets at startup. To react to address changes, it would have to monitor changes via netlink, etc

Note that for `--nodeport-addresses ""` or `--nodeport-addresses 0.0.0.0/0` we just listen on `0.0.0.0` so specific IPs don't matter. And for a more-specific `--nodeport-addresses` value, you're presumably telling it to listen on a specific address that's expected to be static, so...

Reply:
I don't mean dynamically changing the ports that are already open, that is much effort for small win.

Just for new created ports only, I don't think is a big deal, but since we already have the good abstractions, this is much easier now